pilot/pkg/leaderelection/leaderelection.go

```diff
@@ -75,7 +75,9 @@ type LeaderElection struct {
 
 // Run will start leader election, calling all runFns when we become the leader.
 func (l *LeaderElection) Run(stop <-chan struct{}) {
-	go l.defaultWatcher.Run(stop)
+	if l.prioritized {
+		go l.defaultWatcher.Run(stop)
```

Comment:
very nit: should we check `l.defaultWatcher != nil`? This is for sure correct, but a change somewhere else could make the logic slightly off in the future?

Reply:
Updated to also check on l.defaultWatcher is not nil.